### PR TITLE
Modernize deprecated setSystemUiVisibility(int) and related flags

### LINKED ISSUE #800: Modernize deprecated setSystemUiVisibility(int) and related flags in View
SystemUiVisibility flags are deprecated. Use [WindowInsetsController](https://developer.android.com/reference/android/view/WindowInsetsController) instead:

- SYSTEM_UI_FLAG_LAYOUT_STABLE: Use [WindowInsets#getInsetsIgnoringVisibility(int)](https://developer.android.com/reference/android/view/WindowInsets#getInsetsIgnoringVisibility(int)) instead to retrieve insets that don't change when system bars change visibility state.

- SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION | SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN: For floating windows, use [LayoutParams#setFitInsetsTypes(int)](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#setFitInsetsTypes(int)) with [Type#navigationBars()](https://developer.android.com/reference/android/view/WindowInsets.Type#navigationBars()). For non-floating windows that fill the screen, call [Window#setDecorFitsSystemWindows(boolean)](https://developer.android.com/reference/android/view/Window#setDecorFitsSystemWindows(boolean)) with false.

- SYSTEM_UI_FLAG_HIDE_NAVIGATION: Use [WindowInsetsController#hide(int)](https://developer.android.com/reference/android/view/WindowInsetsController#hide(int)) with [Type#navigationBars()](https://developer.android.com/reference/android/view/WindowInsets.Type#navigationBars()) instead.

- SYSTEM_UI_FLAG_FULLSCREEN: Use [WindowInsetsController#hide(int)](https://developer.android.com/reference/android/view/WindowInsetsController#hide(int)) with [Type#statusBars()](https://developer.android.com/reference/android/view/WindowInsets.Type#statusBars()) instead.

- SYSTEM_UI_FLAG_IMMERSIVE_STICKY: Use [WindowInsetsController#BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE](https://developer.android.com/reference/android/view/WindowInsetsController#BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE) instead.

## Location
https://github.com/Igalia/wolvic/pull/787/files#r1251790378

## Possible Solution
As suggested in:
- https://developer.android.com/reference/android/view/View#setSystemUiVisibility(int)
- https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LAYOUT_STABLE
- https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
- https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
- https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_HIDE_NAVIGATION
- https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_FULLSCREEN
- https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_IMMERSIVE_STICKY

### PR DESCRIPTION
**Pull request by HollowMan6**
Resolves #800

https://stackoverflow.com/a/65128718

### CODE REVIEW

File: app/src/common/shared/com/igalia/wolvic/ui/OffscreenDisplay.java [hunk 1]
```diff
@@ -123,17 +126,24 @@ public void onCreate(Bundle savedInstanceState) {
...
-                                View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-                                        | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-                                        | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-                                        | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-                                        | View.SYSTEM_UI_FLAG_FULLSCREEN
-                                        | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                    getWindow().setDecorFitsSystemWindows(false);
+                    WindowInsetsController controller = getWindow().getInsetsController();
+                    if(controller != null) {
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/OffscreenDisplay.java:132]: Nit: space after the if

File: app/src/common/shared/com/igalia/wolvic/ui/OffscreenDisplay.java [hunk 2]
```diff
@@ -123,17 +126,24 @@ public void onCreate(Bundle savedInstanceState) {
...
+                    getWindow()
+                            .getDecorView()
+                            .setSystemUiVisibility(
+                                    View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                                            | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                                            | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                                            | View.SYSTEM_UI_FLAG_FULLSCREEN
+                                            | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+                }
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/OffscreenDisplay.java:146]: Correct me if I'm missing something but can't we just simply call `SetImmersiveSticky()`  here and avoid this code duplication?

**Review by svillar — CHANGES_REQUESTED** [review]

File: app/src/common/shared/com/igalia/wolvic/ui/OffscreenDisplay.java (hunk 2 above)
Comment (HollowMan6) [reply app/src/common/shared/com/igalia/wolvic/ui/OffscreenDisplay.java:146]: I'm not sure how we can re-use the `SetImmersiveSticky()`. By the way, the code duplication exists in the original code and is not introduced by this PR.

Comment (svillar) [reply app/src/common/shared/com/igalia/wolvic/ui/OffscreenDisplay.java:146]: Forget about that, I didn't realize that the other was in the offscreen display code, I somehow assumed that everything was inside the PlatformActivity

**Review by svillar — APPROVED** [review]

**Pull request merged**